src/Fields/broadcast.jl

```diff
@@ -32,20 +32,57 @@ Base.Broadcast.broadcastable(field::Field) = field
 Base.eltype(bc::Base.Broadcast.Broadcasted{<:AbstractFieldStyle}) =
     Base.Broadcast.combine_eltypes(bc.f, bc.args)
 
+# to_struct: for recursively converting a point field into a struct
+to_struct(x::DataLayouts.DataF{T}) where {T <: Geometry.XYPoint} =
+    T(parent(x)...)
+to_struct(x::DataLayouts.DataF{T}) where {T <: Real} = T(first(parent(x)))
+to_struct(x::DataLayouts.DataF{T}) where {T <: StaticArrays.SMatrix} =
+    T(parent(x)...)
+to_struct(x::DataLayouts.DataF{T}) where {T <: Geometry.AxisTensor} =
+    T(axes(T), to_struct.(getproperty.(Ref(x), propertynames(x)))...)
+to_struct(lg::DataLayouts.DataF{T}) where {T <: Geometry.LocalGeometry} =
+    T((to_struct.(getproperty.(Ref(lg), propertynames(lg))))...)
+
+outer_to_struct(::Type{FT}) where {FT <: Real} = FT
+outer_to_struct(x::Spaces.PointSpace) = to_struct(x.local_geometry)
+outer_to_struct(x) = x
+
+_first(field::Field) = _first(axes(field))
```

Comment:
don't you want the values?

Reply:
Ahh, good point. Crud, this may ruin my dispatch pattern

Reply:
Oh, wait, I'm actually not testing with this. This will open happen when we are creating one field from another, which I've not seen too frequently, but it would still be nice to cover this case.

Reply:
I've fixed a bunch of broken things